file to folder and index (#3355)
bbc/simorgh: cypress/integration/README.md

# How our Cypress tests work

> We have a very bespoke approach to testing. We split our tests in 3 sections: Application, Pages and specialFeatures (features that cannot be tested in a single page visit). 

Author: ```suggestion
We have a very bespoke approach to testing. We split our tests in 3 sections: Application, Pages and Special Features (features that cannot be tested in a single page visit). 
```

> We only run a subset of these integrations tests in CI and CD (those we always run are called smoke tests), but have a regular [cron job](https://en.wikipedia.org/wiki/Cron) that runs all e2e tests in this directory against localhost, test.bbc.com and www.bbc.com.

Author: ```suggestion
We only run a subset of these integrations tests in CI and CD (those we always run are called smoke tests), but have a regular [cron job](https://en.wikipedia.org/wiki/Cron) that runs all integration tests (a.k.a. e2e tests) in this directory against localhost, test.bbc.com and www.bbc.com.
```

## [Config used in the tests](../support/config)

> It's important to familiarise yourself with the service.js config before writing e2e tests. Within there we define which combinations of services and pageTypes should be tested, what the paths to test on are and whether something should be tested while only smoke testing.

Author: ```suggestion
It's important to familiarise yourself with the service.js config before writing e2e tests. Within there we define which combinations of services and page types should be tested, what the paths to test on are and whether something should be tested while only smoke testing.
```

> ## Application

Author: ```suggestion
## [Application](./application)
```

Here we test things that aren't pages such as manifest files and service workers. NB we don't test the .json endpoint for each page here because it's an implicit part of every 'page' test.

> ## Pages

Author: ```suggestion
## [Pages](./pages)
```

This is where the overwhelming majority of e2es live. 
Key points:
 - Files in the base of the directory run on all page types.

>  - There is a directory per page type where each page's custom test are written.

Author: ```suggestion
 - There is a directory per page type where each page's custom tests are written.
```

>  - The index.js of each page type directory must use the [runTestsForPage function](../support/runTestsForPage.js), this reduces the number of page visits to 1 visit to the canonical page and 1 visit to the amp version of the same page.

Author: ```suggestion
 - The index.js of each page type directory must use the [runTestsForPage function](../support/runTestsForPage.js), this reduces the number of page visits to one visit to the canonical page and one visit to the [AMP](https://amp.dev/) version of the same page.
```

> ## Special Features

Author: ```suggestion
## [Special Features](./specialFeatures)
```
